src/cascadia/TerminalSettingsEditor/SettingContainer.cpp

```diff
@@ -172,12 +144,82 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
     }
 
-    hstring SettingContainer::_GenerateOverrideMessageText()
+    void SettingContainer::_UpdateOverrideMessage()
     {
-        if (HasSettingValue())
+        if (const auto& child{ GetTemplateChild(L"ResetButton") })
         {
-            return hstring{ fmt::format(std::wstring_view{ RS_(L"SettingContainer_OverrideIntro") }, RS_(L"SettingContainer_OverrideTarget")) };
+            if (const auto& button{ child.try_as<Controls::Button>() })
+            {
+                if (HasSettingValue())
+                {
+                    // We want to be smart about showing the override system.
+                    // Don't just show it if the user explicitly set the setting.
+                    // If this flag is set, we'll hide the entire override system.
+                    bool hideOverrideSystem = false;
+
+                    const std::wstring templateText{ RS_(L"SettingContainer_OverrideIntro") };
+                    const auto pos{ templateText.find(L"{}") };
+                    std::wstring tooltip{};
+                    if (pos != std::wstring::npos)

```

Comment:
Yeah, I wasn't sure what to do here. The only time this would be false is if the localized text didn't have `{}`. Figured, if that happens, we'd want a generic message to fallback to.